controllers/xsite_config.go

```diff
@@ -120,44 +120,60 @@ func appendKubernetesRemoteLocation(ctx context.Context, infinispan *ispnv1.Infi
 	remoteServiceName := infinispan.GetRemoteSiteServiceName(remoteLocationName)
 	remoteRouteName := infinispan.GetRemoteSiteRouteName(remoteLocationName)
 
-	var host string
-	var port int32
-	var err error
-
-	// Try to find the service first
-	siteService := &corev1.Service{}
-	err = remoteKubernetes.Client.Get(ctx, types.NamespacedName{Name: remoteServiceName, Namespace: remoteNamespace}, siteService)
+	routeObj := &reconcileType{
+		ObjectType:   &routev1.Route{},
+		GroupVersion: routev1.SchemeGroupVersion,
+	}
+	routeSupported, err := remoteKubernetes.IsGroupVersionSupported(routeObj.GroupVersion.String(), routeObj.Kind())
 	if err != nil {
-		logger.Error(err, "could not get x-site service in remote cluster", "site service name", remoteServiceName, "site namespace", remoteNamespace)
-		if !errors.IsNotFound(err) {
-			// another error
-			return err
-		}
+		logger.Error(err, fmt.Sprintf("failed to check if GVK '%s' is supported", routeObj.GroupVersionKind()))
+		return err
 	}
 
-	// service found
-	if err == nil {
-		host, port, err = getCrossSiteServiceHostPort(siteService, remoteKubernetes, logger, eventRec, "XSiteRemoteServiceUnsupported", ctx)
-		if err != nil {
-			logger.Error(err, "error retrieving remote x-site service information")
-			return err
-		}
-	} else {
-		// Try searching for the Route
+	if routeSupported {
```

Comment:
Wouldn't it be a bit clearer if we checked that expose type for cross site is set to `Route` before blindly attempting to get the hostname just because Route is supported?

Reply:
in theory, you can have one cluster exposing cross-site as a Route and another as a LoadBalancer (or NodePort) service.